static/stripe.html

```diff
@@ -0,0 +1,544 @@
+                    stripe.confirmCardPayment(clientSecret).then(function(newConfirmResult) {
+                        if (newConfirmResult.error) {
+                            // The payment failed -- TODO ask user for a new payment method?
+                            showError($('#payment-form'), newConfirmResult.error.message);
+                        } else {
+                            // The payment has succeeded.
+                            orderComplete(newConfirmResult.paymentIntent)
+                        }
+                    });
+                } else {
+                    // The payment has succeeded.
+                    orderComplete(confirmResult.paymentIntent)
+                }
+            }
+        });
+    };
+
+    // Calls stripe.confirmCardPayment
+    // If the card requires authentication Stripe shows a pop-up modal to
+    // prompt the user to enter authentication details without leaving your page.
+    var payWithCard = function(stripe, card, clientSecret) {
+        loading($('#payment-form'), true);
+        stripe.confirmCardPayment(
+            clientSecret,
+            {
+                payment_method: {
+                    card: card
+                }
+            }
+        ).then(function(confirmResult) {
+            if (confirmResult.error) {
+                // Show error to your customer
+                showError($('#payment-form'), confirmResult.error.message);
+            } else {
+                // The payment succeeded!
+                orderComplete(confirmResult.paymentIntent);
+            }
+        });
+    };
+
+    /* ------- UI helpers ------- */
+
+    // Shows a success message when the payment is complete
+    var orderComplete = function(payment) {
+        loading($('#payment-form'), false);
+        loading($('.result-message'), true)
+        $("button[type=submit]").attr('disabled', true)
+        $('#payment-form').addClass('hidden')
+        $('.result-message .success').addClass('hidden')
+        $('.result-message .error').addClass('hidden')
+
+        $('#payment-id').text(payment['id'])
+        $('.result-message').removeClass('hidden')
+        api.redeemPayment(payment['id'], payment['receipt_email'])
+            .then(function (token) {
+                $('#token').text(token)
+                $('#register').attr('href', '/register?token='+encodeURIComponent(token))
+                loading($('.result-message'), false)
+                $('.result-message .error').addClass('hidden')
+                $('.result-message .success').removeClass('hidden')
+            })
+            .catch(function (error) {
+                // TODO handle some errors gracefully e.g. amount too small
+                loading($('.result-message'), false)
+                $('.result-message .success').addClass('hidden')
+                $('.result-message .error')
+                    .removeClass('hidden')
+                    .text(error)
```

Comment:
Probably shouldn't forget to do this - the user being told "error too small" when they donate is not good lmao.

Thinking about it, the only reason we need to "redeem" the stripe payment when `amount < $5` is to log the donation to discord; if we start doing those logs via a webhook then the client doesn't have any reason to call `/stripe/redeem`...